src/items/skill-item/skillSheet.ts

```diff
@@ -108,4 +111,179 @@ export class SkillSheet extends RqgItemSheet<ItemSheet.Options, SkillSheetData |
       ui.notifications?.warn("You can only drop a journalEntry");
     }
   }
+
+  static showImproveSkillDialog(actor: RqgActor, skillItemId: string, speakerName: string): void {
+    const item = actor.items.get(skillItemId);
+    if (!item || item.data.type !== ItemTypeEnum.Skill) {
+      const msg = `Couldn't find skill with itemId [${skillItemId}] on actor ${actor.name} to show Improve Skill dialog.`;
+      ui.notifications?.error(msg);
+      throw new RqgError(msg);
+    }
+    const skill = item.data as SkillDataProperties;
+    
+    if (item.data.data.chance > 75) {
+      //@ts-ignore hideTraining
+      item.data.data.hideTraining = true;
+    }
+
+    const chance: number = Number(item.data.data.chance) || 0;
+    const categoryMod: number = Number(item.data.data.categoryMod) || 0;
+    var chanceToGain = 0;
+    if (chance > 100) {
+      chanceToGain = categoryMod;
+    } else {
+      chanceToGain = (100 - chance) + categoryMod;
+    }
+
+    const btnImprove = getGame().i18n.format("DIALOG.improveSkillDialog.btnDoImprovement");
+    const btnCancel = getGame().i18n.format("DIALOG.improveSkillDialog.btnCancel");
+    var buttons = {};
```

Comment:
Just some syntax, but it you type this as `const buttons: any = {}` then you don't need the @ts-ignores further down.

Also generally, please don't use `var`. Use `const` wherever possible and `let` if that doesn't compile to keep variable scopes to a minimum :-)

Reply:
The limits of my javascript and typescript knowledge are showing!